src/game/client/baseheightmap.cpp

```diff
@@ -968,86 +968,88 @@ void BaseHeightMapRenderObjClass::Do_The_Light(VertexFormatXYZDUV2 *vb,
+                interpolated_height = height_4;
             } else {
-                float f10 = data[x_extent + i3];
-                f9 = ((1.0f - f6) * (f7 - f10) + f10 + (f8 - f10) * f5) * HEIGHTMAP_SCALE;
+                float height_3 = data[x_extent + index_1];
+                float height_5 = ((1.0f - fract_y) * (height_1 - height_3) + height_3 + (height_2 - height_3) * fract_x) * HEIGHTMAP_SCALE;
+                interpolated_height = height_5;
             }
 
+            // Calculate the normal vector if pos is provided
             if (pos != nullptr) {
-                int i4 = x_extent * (i2 - 1) + i1;
-                int i5 = x_extent * i2 + i1;
-                int i6 = x_extent + i5;
-                int i7 = x_extent * (i2 + 2) + i1;
-                unsigned char c1 = data[i5];
-                unsigned char c2 = data[i5 + 1];
-                unsigned char c3 = data[x_extent + 1 + i5];
-                unsigned char c4 = data[x_extent + i5];
-                unsigned char c5 = data[i4];
-                unsigned char c6 = data[i4 + 1];
-                unsigned char c7 = data[i7 + 1];
-                unsigned char c8 = data[i7];
-                float f11 = c2 - data[i5 - 1];
-                float f12 = data[i5 + 2] - c1;
-                float f13 = data[x_extent + 2 + i5] - c4;
-                float f14 = c4 - c5;
-                float f15 = c3 - c6;
-                float f16 = c7 - c2;
-                float f17 = c8 - c1;
-                float f18 = (1.0f - f5) * f11 + f5 * f12;
-                float f19 = (1.0f - f5) * f12 + f5 * f13;
-                float f20 = f18 * (1.0f - f6) + f6 * f19;
-                float f21 = (1.0f - f5) * f14 + f5 * f17;
-                float f22 = (1.0f - f5) * f15 + f5 * f16;
-                float f23 = f21 * (1.0f - f6) + f6 * f22;
-                Vector3 v1;
-                Vector3 v2;
-                Vector3 v3;
-                v1.Set(32.0f, 0.0f, f20);
-                v2.Set(0.0f, 32.0f, f23);
-                Vector3::Normalized_Cross_Product(v1, v2, &v3);
-                pos->x = v3.X;
-                pos->y = v3.Y;
-                pos->z = v3.Z;
+                int index_2 = x_extent * (index_y - 1) + index_x;
+                int index_3 = x_extent * index_y + index_x;
+                int index_4 = x_extent + index_3;
+                int index_5 = x_extent * (index_y + 2) + index_x;
+                unsigned char height_6 = data[index_3 + 1];
+                unsigned char height_7 = data[index_4 + 1];
+                unsigned char height_8 = data[index_4];
+                float diff_1 = height_2 - height_3;
+                float diff_2 = height_3 - height_1;
+                float diff_3 = height_8 - height_2;
+                float diff_4 = height_2 - height_5;
+                float diff_5 = height_3 - height_6;
+                float diff_6 = height_7 - height_2;
+                float interpolated_diff_x = (1.0f - fract_x) * diff_

```

Comment:
Despite this not compiling now, I don't like how this code was refactored beyond the name changes. it is difficult to tell whether or not it will behave the same.